client/src/context/notificationContext.tsx

```diff
@@ -0,0 +1,31 @@
+import { FunctionComponent, createContext, useState, useEffect } from 'react';
+import { getNotification } from '../helpers/APICalls/notification';
+import { Notification } from '../interface/User';
+
+interface NotificationContext {
+  notifications?: Notification[];
+  setId: (id: number) => void;
+}
+
+export const NotificationContext = createContext<NotificationContext>({
+  notifications: [],
+  setId: () => null,
+});
+
+export const NotificationProvider: FunctionComponent = ({ children }): JSX.Element => {
+  const [notifications, setNotifications] = useState<Notification[]>();
+  const [id, setId] = useState<number>();
```

Comment:
What is the purpose of this state? :)

Reply:
Basically, I used it to fetch the notifications and update the UI. When someone clicks on the delete button and if the notification is deleted, I wanted to update the UI without reloading the page. That's why used this 'id' as a dependency for the useEffect to fetch the notifications. I hope it makes sense :)
May be in the frontend i should only use it for the 'See all' page which contains the delete button, but not in the notification pop-up box

Reply:
That makes sense, thanks for the clarification. 

On a side note: I guess it would be much faster to filter through the current state and exclude the notification with the specific Id as opposed to retrieving all the notifications each time :)

Reply:
removed the fetching based on id in the last commit and did some clean up.